### ISSUE TITLE
Identify and fix source of randomness in transformer initialization

### ISSUE THREAD

**Issue by aliciafmachado** [PAIR-code/tiny-transformers#35]
Currently, if you run `animated-transformer/src/lib/trainer/basic_transformer_trainer.spec.ts` two times in a roll, the initial loss does not match. (before the sgd step).

The task seems to be generating reproducible results, and the neural network initialized parameters are also reproducible. However, once you initialize the train state, the loss is different on every run.

**Comment by aliciafmachado** [comment]
Token embedder was not properly initialized. Will send a PR to fix it.
